# --- SDE-COPYRIGHT-NOTE-BEGIN ---
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
#
# Filename: package/.../spqmail/0004-qmail-sasl-auth-debug-log-no-password.patch
# Copyright (C) 2012 The OpenSDE Project
#
# More information can be found in the files COPYING and README.
#
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License as published by the Free Software
# Foundation; either version 2 of the License, or (at your option) any later
# version.
# --- SDE-COPYRIGHT-NOTE-END ---

From f4754d2929bef3a6ae3e1cb59eb01b8c6376e9cb Mon Sep 17 00:00:00 2001
From: Christian Wiese <christian.wiese@securepoint.de>
Date: Tue, 20 Nov 2012 12:01:45 +0100
Subject: [PATCH] qmail-sasl.c: changed to only log passwords if
 AUTH_DEBUG_PASSWORD is set

---
 qmail-sasl.c |   10 ++++++++--
 1 file changed, 8 insertions(+), 2 deletions(-)

diff --git a/qmail-sasl.c b/qmail-sasl.c
index 76d8215..b290495 100644
--- a/qmail-sasl.c
+++ b/qmail-sasl.c
@@ -63,7 +63,10 @@ int authenticate(int method)
 
   if (env_get("DEBUG_AUTH")) {
     Log("SASL user:");Log(user.s);Log("\n");
-    Log("SASL pass:");Log(pass.s);Log("\n");
+    /* only log passwords if AUTH_DEBUG_PASSWORD is set */
+    if (env_get("DEBUG_AUTH_PASSWORD")) {
+      Log("SASL pass:");Log(pass.s);Log("\n");
+    }
     Log("SASL resp:");Log(resp.s);Log("\n");
   }
   flush();
@@ -203,7 +206,10 @@ int auth_plain(char *arg)
   }
 
   r_pass = slop.s + i;
-  doname(); Log("SASL pass("); Log(r_pass); Log(")\n"); flush();
+  /* only log passwords if AUTH_DEBUG_PASSWORD is set */
+  if (env_get("AUTH_DEBUG_PASSWORD")) {
+    doname(); Log("SASL pass("); Log(r_pass); Log(")\n"); flush();
+  }
   if (!stralloc_copys(&pass, r_pass)) die_nomem();
   while (slop.s[i++]) {
     if (i >= slop.len) {
-- 
1.7.10.2


From b30da23cae39be0bdd648af8a92190f13db01022 Mon Sep 17 00:00:00 2001
From: Christian Wiese <christian.wiese@securepoint.de>
Date: Tue, 20 Nov 2012 13:00:25 +0100
Subject: [PATCH] qmail-sasl.c: changed auth_plain() to only log base64
 encoded credentials if AUTH_DEBUG_PASSWORD is set

---
 qmail-sasl.c |   10 +++++++++-
 1 file changed, 9 insertions(+), 1 deletion(-)

diff --git a/qmail-sasl.c b/qmail-sasl.c
index b290495..ce91fe3 100644
--- a/qmail-sasl.c
+++ b/qmail-sasl.c
@@ -171,7 +171,15 @@ int auth_plain(char *arg)
   int i=0;
   char *r_pass, *r_user, x[123];
 
-  doname(); Log("SASL 1("); Log(arg); Log(")\n"); flush();
+  doname(); Log("SASL 1(");
+  /* only log base64 encoded credentials if AUTH_DEBUG_PASSWORD is set */
+  if (env_get("AUTH_DEBUG_PASSWORD")) {
+    Log(arg);
+  } else {
+    Log("credentials hidden");
+  }
+  Log(")\n"); flush();
+
   if (*arg) {
     if (b64decode(&slop,arg,str_len(arg)) != 0)
       return ERR_AUTH_INPUT;
-- 
1.7.10.2

